### PR TITLE
(PDB-2617) stale-nodes: use newest report/catalog

### PR DESCRIPTION
**Pull request by rbrw**
Don't consider a node stale just because any report/catalog is too old;
require that the newest report/catalog be too old.

### CODE REVIEW

**Comment by wkalt** [comment]
It seems like this will never expire nodes for which no catalog exists?

**Comment by senior** [comment]
@rbrw @wkalt I think what's there should work. Seems like we should have a test for the original bug and a test for this as well though

**Comment by rbrw** [comment]
@senior agreed, I'll see about adding some.

**Comment by wkalt** [comment]
@senior @rbrw yeah my comment was wrong. this looks fine.

**Comment by ajroetker** [comment]
👍

File: test/puppetlabs/puppetdb/scf/storage_test.clj [hunk 1]
```diff
@@ -1229,14 +1240,73 @@
...
-                     (assoc :environment "ENV2")
-                     (assoc :end_time (now))
-                     (assoc :producer_timestamp (now)))]
-      (store-example-report! report (now))
-      (is (= (stale-nodes (-> 1 days ago)) [])))))
+(deftest stale-nodes-behavior-for-reports
+  (let [report-at #(-> (:basic reports)
+                       (assoc :environment "ENV2")
+                       (assoc :end_time %)
+                       (assoc :producer_timestamp %))
```
Comment (ajroetker) [inline test/puppetlabs/puppetdb/scf/storage_test.clj]: Minor nitpick but you only need 1 `assoc` call here

Comment (rbrw) [inline test/puppetlabs/puppetdb/scf/storage_test.clj]: Uhh, right...

**Pull request merged**